src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -693,15 +706,19 @@ export const handleGetEncryptedResponse: RequestHandler<
  * @returns 422 when user in session cannot be retrieved from the database
  * @returns 500 if any errors occurs whilst querying database
  */
-export const handleGetMetadata: RequestHandler<
+export const getMetadata: RequestHandler<
   { formId: string },
   SubmissionMetadataList | ErrorDto,
   unknown,
-  Query & { page?: number; submissionId?: string }
+  Query &
+    RequireAtLeastOne<
+      { page?: number; submissionId?: string },
+      'page' | 'submissionId'
+    >
 > = async (req, res) => {
   const sessionUserId = (req.session as Express.AuthedSession).user._id
   const { formId } = req.params
-  const { page, submissionId } = req.query
+  const { page = 1, submissionId } = req.query
```

Comment:
can we restrict the scope of this PR to the API refactor only? let's not repeat the mistakes of the previous production bug by mixing refactors and logic changes

Reply:
sure, i'll revert this change to adhere to before